src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -107,6 +229,39 @@ InvokerFactory MakeImplGemmDynamicDataInvokerFactory(const ConvolutionContext& c
             };
         };
     }
+    else if(ctx.direction.IsBackwardData())
+    {
+        return [ctx](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const boost::any& primitive_parameters) {
```

Comment:
Here too, pass variables as reference return[&]

Reply:
I am not sure. @DrizztDoUrden - ?

Reply:
No, because it would get destroyed at the run stage or even earlier.

Reply:
@DrizztDoUrden Are you sure about it?  If you have access to Handle, why would you loose ctx object?  are Handles owned outside ctx?

Reply:
Yes. I mean, I have taken a significant part in designing the system of solvers and conv ctx, but you shouldn't take my word for granted anyway. I would suggest everyone making solvers at some point look where we create them and their lifecycle in general.
TL;DR: The context is created inside of find and is never stored between API calls. It contains data that may become outdated between calls. The most simple example is the workspace size and all the buffers.

Reply:
This capture of ctx is changed to conv::ProblemDescription